ambry-commons/src/main/java/com.github.ambry.commons/BlobIdBuilder.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.commons;
+
+import com.github.ambry.account.Account;
+import com.github.ambry.account.Container;
+import com.github.ambry.clustermap.PartitionId;
+
+
+/**
+ * A builder class for {@link BlobId}. This class is not thread safe.
+ */
+public class BlobIdBuilder {
```

Comment:
Had a discussion with @vgkholla . We feel that builder is not required here. Especially when we have two properly defined constructors and have mandatory fields. Can we remove the builder unless you have strong reasons to hold on to it.